lib/extensions/map_extensions.dart

```diff
@@ -0,0 +1,7 @@
+/// Extensions for [Map] class that provide additional functionality and convenience methods.
+extension CustomerIOMapExtension on Map<String, dynamic> {
+  /// Returns a new map with entries that have non-null values, excluding null values.
+  Map<String, dynamic> excludeNullValues() {
+    return Map.fromEntries(entries.where((entry) => entry.value != null));
```

Comment:
```suggestion
    return  map.removeWhere((key, value) => value == null);
```
not this exact place but dart provides this utility method

Reply:
It definitely sounds good, but I deliberately avoided modifying the same map because removing items directly could unintentionally alter customer provided map, which might cause issues in their app. So I created a new map instance to avoid any potential side effects.